### ISSUE TITLE
contour locked up when I catted a file of unknown contents

### ISSUE THREAD

**Issue by cqexbesd** [contour-terminal/contour#1062]
### Contour Terminal version

0.3.11-master-cd0aa937

### Installer source

Github: source code cloned

### Operating System

Arch Linux as of approx a week ago

### Architecture

x86-64

### Steps to reproduce

I cat'ed a file and I have no idea whats in there. I will try to save the file if I can but its not as simple as you might think.

### Expected Behavior

something to appear on my screen

### Actual Behavior

contour stopped responding. the mouse cursor disappears when inside the window. The window does redraw however. `strace` says it is constantly reading `WW`...not sure where that's coming from but see attached.

[contour.strace.txt](https://github.com/contour-terminal/contour/files/10947147/contour.strace.txt)
[contour.lsof.txt](https://github.com/contour-terminal/contour/files/10947149/contour.lsof.txt)

### Additional notes

contour is taking over 100% of CPU time, not unsurprisingly.

**Comment by Yaraslaut** [comment]
If you can not save the file that we can use to reproduce this behavior, output of `contour debug all` will help a lot 
Can be related to https://github.com/contour-terminal/contour/issues/980

**Comment by Yaraslaut** [comment]
Here is example of a file which will crash contour
[crash.txt](https://github.com/contour-terminal/contour/files/11235924/crash.txt)

`tail -n 20 crash.txt`  is enough 
also 
`head -n 10 crash.txt` shows weird behavior as well

**Comment by Utkarsh-khambra** [comment]
> tail -n 20 crash.txt is enough
This file doesn't crash contour for me.

**Comment by christianparpart** [comment]
Yeah u started working on this and u think I only fixed the lockup, but what is missing is proper handling of arbitrary binary (invalid utf-8) data. This requires sissy a Twitter in libunicode sich I didn't finish yet
